Explorer/Assets/DCL/Backpack/BackpackGridController.cs

```diff
@@ -0,0 +1,157 @@
+using Arch.Core;
+using Arch.SystemGroups;
+using Cysharp.Threading.Tasks;
+using DCL.AssetsProvision;
+using DCL.AvatarRendering.Wearables.Components;
+using DCL.AvatarRendering.Wearables.Components.Intentions;
+using DCL.Backpack.BackpackBus;
+using DCL.Web3Authentication.Identities;
+using ECS.Prioritization.Components;
+using ECS.StreamableLoading.Common;
+using System.Collections.Generic;
+using System.Threading;
+using UnityEngine;
+using UnityEngine.Pool;
+using Utility;
+using Object = UnityEngine.Object;
+using ParamPromise = ECS.StreamableLoading.Common.AssetPromise<DCL.AvatarRendering.Wearables.Components.IWearable[], DCL.AvatarRendering.Wearables.Components.Intentions.GetWearableByParamIntention>;
+
+namespace DCL.Backpack
+{
+    public class BackpackGridController
+    {
+        private readonly BackpackGridView view;
+        private readonly BackpackCommandBus commandBus;
+        private readonly BackpackEventBus eventBus;
+        private readonly IWeb3IdentityCache web3IdentityCache;
+        private readonly NftTypeIconSO rarityBackgrounds;
+        private readonly NFTColorsSO rarityColors;
+        private readonly NftTypeIconSO categoryIcons;
+
+        private IObjectPool<BackpackItemView> gridItemsPool;
+        private readonly Dictionary<string, BackpackItemView> usedPoolItems;
+        private World world;
+
+        public BackpackGridController(
+            BackpackGridView view,
+            BackpackCommandBus commandBus,
+            BackpackEventBus eventBus,
+            IWeb3IdentityCache web3IdentityCache,
+            NftTypeIconSO rarityBackgrounds,
+            NFTColorsSO rarityColors,
+            NftTypeIconSO categoryIcons)
+        {
+            this.view = view;
+            this.commandBus = commandBus;
+            this.eventBus = eventBus;
+            this.web3IdentityCache = web3IdentityCache;
+            this.rarityBackgrounds = rarityBackgrounds;
+            this.rarityColors = rarityColors;
+            this.categoryIcons = categoryIcons;
+
+            usedPoolItems = new ();
+            eventBus.EquipEvent += OnEquip;
+            eventBus.UnEquipEvent += OnUnequip;
+        }
+
+        public void InjectToWorld(ref ArchSystemsWorldBuilder<World> builder, in Entity playerEntity)
+        {
+            world = builder.World;
+        }
+
+        public async UniTask InitialiseAssetsAsync(IAssetsProvisioner assetsProvisioner, CancellationToken ct)
+        {
+            BackpackItemView backpackItem = (await assetsProvisioner.ProvideInstanceAsync(view.BackpackItem, ct: ct)).Value;
```

Comment:
If you call `assetsProvisioner.ProvideInstanceAsync` it will make an instance. You rather need an original asset to create a pool from here, don't you?